Feature book review edit

wsh096 opened the pull request:
book과 리뷰를 수정 하는 브랜치.

wsh096 committed:
feat : ✨ ResponseCode 응답값 기본 포맷.

1. ApiResponse 구현

2. code 구현, errorCode msg로 수정.(interface 구현)

3. 남은 할 일. API 명세서 대로 값 만들기.

wsh096 committed:
feat : ✨ 회원가입 response 응답값 구현.

footer - fix : 🚑 BookController id,desc로 sort 기본값 설정.

하지만, id만 입력시, 아무것도 안 넘어간다는 에러 메시지가 발생함 해당 로직 처리가 필요

wsh096 committed:
feat : ✨ 회원가입, 로그인 response 응답값 구현.

wsh096 committed:
fix : 🚑 로그인 response 응답값 TreeMap 으로 변경

wsh096 committed:
fix : 🚑 OAuthLogin Response 응답값 구현.

-> URI도 API 명세서 기준으로 수정.

wsh096 committed:
init: 🎉 userLogOut 기능 관련 기본 내용 커밋.

1. Redis 등 설정해 줘야 할 부분이 많아 남겨둠.

2. 포맷 먼저 잡고, JWT 등의 요소들이 바뀌어야 할 것으로 보임.
현재 토큰 1개, 추후 토큰 2개

wsh096 committed:
init: 🎉

wsh096 committed:
feat : ✨ userLogOut 기능 응답값 만들기.

wsh096 committed:
feat : ✨ 프론트와 통신하는 CORS 설정

wsh096 committed:
fix : 🚑 OAuth loginResponse, GET으로 하지 않으면, 저장이 되지 않음(공식문서도 GET 방식으로 하라고 지정)

이와 추가적으로 nickname저장 형태 구현.

wsh096 committed:
fix : 🚑 OAuthController 수정

wsh096 committed:
fix : 🚑 OAuth 기능 수정 완료

프론트와 소통 필요

wsh096 committed:
feat : ✨ 리다이렉트 주소 형태 구현.
OAuth에 값 넣어서 프론트로 넘겨주는 방식.

footer : OAuthController 줄 바꿈 수정은 부수적

wsh096 committed:
fix : 🚑 리다이렉트 검증 완료.

토큰이 정상적으로 넘어가게 값을 입력 완료.

footer : 주소의 경우, 임시적으로 넣어둔 거라 추후 변경 가능

wsh096 committed:
fix : 🚑 Jwt 토큰 암호화 데이터 저장 하지 않도록 변경

1. 회의 결과, 토큰을 디코딩해 사용할 것이며, 필요한 값은 이메일, 닉네임, userId로 사용

wsh096 committed:
fix : 🚑 뎁스 통일

pumkinni committed:
feature : ✨ RoomPicture Entity 추가 및 room entity 위치 수정

room entity accommodation entity 안으로 이동

book controller, repository, service
Review controller, service 주석 처리
- entity 수정 후 충돌 방지 (추후 수정 필요)

wsh096 committed:
feat : ✨ QueryDsL 의존성 추가

footer fix : 🚑 뎁스 통일

book 관련된 가격 정보 및 Price 수정.
필요 레퍼지토리 주입.

wsh096 committed:
fix : 🚑 QueryDsL 실행 관련 요소 해결

1. mappedBy로 인한 conflict 해결

    @ManyToOne
    @JoinColumn(name = "appUser_id")
    private AppUser user;

wsh096 committed:
fix : 🚑 TestFile 에러 외래키 데이터 무결성 에러.

현재 관련 내용 수정하지 않고 이후로 넘김.

wsh096 committed:
fix : bookService list 반환

1. [프론트 요청] page 제거 완료.

2. [프론트 요청] 결과값 반환을 위한 list 생성. //응답값 형태 변경은 아직 변경 안 함.

3. [review] 관련 > 리뷰 작성 시 평점 변경 현재 값에 평점 새로 변환해서 하는 형태로 로직 형성.

4. 모든 변수명 null 값 허용하는 object로 느슨하게 조절.

5. config 폴더도 기본적인 global 설정이라는 생각 하에 config 폴더 위치 변경.

6. 현재 예약 결과값을 반환하는 형태는 forEach 형태 즉시 stream으로 바꿀 예정.

wsh096 committed:
fix : bookService list 반환

1. [프론트 요청] page 제거 완료.

2. [프론트 요청] 결과값 반환을 위한 list 생성. //응답값 형태 변경은 아직 변경 안 함.

3. [review] 관련 > 리뷰 작성 시 평점 변경 현재 값에 평점 새로 변환해서 하는 형태로 로직 형성.

4. 모든 변수명 null 값 허용하는 object로 느슨하게 조절.

5. config 폴더도 기본적인 global 설정이라는 생각 하에 config 폴더 위치 변경.

6. 현재 예약 결과값을 반환하는 형태는 forEach 형태 즉시 stream으로 바꿀 예정.

wsh096 committed:
fix : 🚑 bookService for -> stream

wsh096 committed:
fix : 🚑 makeBook update

1. bookForm people 도 객체로 변환 int > Integer

2. 스트림으로 작성. price의 변경은 예약을 시작하기 전, 예약 후에도 적용해 여러 사람이 동시에 접근했을 때 고려.

3. 단 이와 같은 상황에서 엣지케이스가 생길 수 있을 것 같긴 한데, 고민해 보기.

wsh096 committed:
fix : 🚑 makeBook update / 검증 및 에러 해결.(인덱스 사용)

//외래키 제약 조건을 넣어줌으로써, 에러 로그(accommodationId 에러 해결)
-> CREATE INDEX idx_accommodationId ON accommodation(accommodationId);(인덱스 설정)

//외래키들 명시적으로 저장해줘야 null 이 아닌 값으로 입력이 됨을 확인**(6.23 코드 리뷰에 공유)

1. Accommodation 필드 모두, 객체로 변환.

2. 스트림으로 작성. price의 변경은 예약을 시작하기 전, 예약 후에도 적용해 여러 사람이 동시에 접근했을 때 고려.

3. 서비스 정상 작동 예시 데이터 활용 검증 완료.

wsh096 committed:
fix : 🚑 makeReview error 발견 / 검증 및 에러 해결.(인덱스 사용)

0. 현재, book이 생기면 , 리뷰에서 연결되어 있는 외래키들의 연결로 스택오버플로우 발생 해당 요소 수정 필요.

//외래키 제약 조건을 넣어줌으로써, 에러 로그(accommodationId 에러 해결)
-> CREATE INDEX idx_accommodationId ON accommodation(accommodationId);(인덱스 설정)

//외래키들 명시적으로 저장해줘야 null 이 아닌 값으로 입력이 됨을 확인**(6.23 코드 리뷰에 공유)

1. Accommodation 필드 모두, 객체로 변환.

2. 평점 로직이 비정상적인 이유는 기존 데이터에 평점이 있기 때문.

3. 서비스 정상 작동 예시 데이터 활용 검증 완료.

footer : create - drop 시 생성 할 때, fk 에러가 발생했는데, update에서는 발생하지 않음(이유는 무엇일까?)

wsh096 committed:
fix : 🚑 reviewDto로 해결.

외래키를 계속 참조하고 타고 들어가서 StackOverFlow 발생으로 추정.

1. 이를 Dto로 지정해서 넘겨주면 문제가 없음.

footer : 비교를 위해 변수인라인화 하지 않음.

wsh096 committed:
fix : 🚑 reviewList 인라인화

wsh096 committed:
fix : 🚑 줄정리

wsh096 committed:
fix : 🚑 makeBook 로직 수정.

1. 기본적으로 총합이 넘어옴.

wsh096 committed:
fix : 🚑 충돌방지를 위한 패키지 위치 수정

wsh096 committed:
fix : ResponseCode 사용하는 걸로만 정리

wsh096 committed:
fix : 🚑 review requestParam
6.23 오전 회의 결과로 정리

> 리뷰는 최근만 오면 되기 때문에 정렬 별도로 사용 하지 않음

wsh096 committed:
fix : 🚑 score > rate 유지
6.23 오전 회의 결과로 정리

wsh096 committed:
fix : 🚑 score > rate 유지
6.23 오전 회의 결과로 정리

wsh096 committed:
fix : 🚑 jwt 토큰에 대한 학습, 토큰값을 받아오는 방법

1. jwt 토큰에서 invalid 에러가 발생했다.
Do Not Allow가 아닌 invalid는 userDto에서 발생한 에러로
해당 에러는 기존에 암호화했던 코드를 복호화 하는 등의 로직의 검증으로 인해 복잡한 해석이 포함되어 있었고, 이를 캐치하지 못해서 문제가 됐다.

2. 이후에는 SQL 에러가 발생했고, 이는 컬럼의 명칭이 빈번하게 바뀌면서 생긴 문제다.

이는 SQL Query문으로 해결했다.
ALTER TABLE room DROP COLUMN id;
이와 같이 속성 값을 이용해 제거 또는 변경하는 방식을 앞으로도 활용할 예정.

wsh096 committed:
fix : 🚑 jwt 토큰에 대한 학습, 토큰값 관련 변경 로직. boolean 관계.

1. jwt 토큰 로직 관련 36번째 JwtAuthenticationProvider 참고.

2. !역으로 되어 있던 관계, -> !을 푼 관계로 변경.(모두 적용 완료)

3. Aes256Utils 사용하지 않아 삭제.

4.Accommodation 관련 controller/service 필요하지 않아 삭제

FeelingXD commented on src/main/java/com/zerobase/yogizogi/global/config/CorsConfig.java:
```diff
@@ -0,0 +1,18 @@
+    public void addCorsMappings(CorsRegistry registry) {
+        registry.addMapping("/**")
+            .allowedOrigins("*")
+            .allowedMethods("GET","POST","PUT","DELETE")
```
해당부분에 PATCH 등 기타 HTTP 메서드가 제외되어있습니다. 추후사용시 문제가될수있으므로 팀 API 스펙을 확인하여 변경이 필요할수있습니다. :)

wsh096 replied:
현재 로직 상에서 GET POST DELETE PATCH만 사용이 될 것 같아 수정이 될 것 같습니다. :) 
프론트와 통신에서 문제가 될 수 있다고 멘토링 받으면서 이야기 되었는데 해당 요소에서 지민님도 이런 문제가 있진 않으셨을까 싶네요.
HTTP 메서드가 정확히 어떤 건지 이해하지 못해 찾아 보고 해당 요소를 반영할 수 있도록 하겠습니다!

FeelingXD commented on src/main/java/com/zerobase/yogizogi/user/controller/UserLogInController.java:
```diff
@@ -10,7 +10,7 @@
 
 @RestController
-@RequestMapping("/users")
+@RequestMapping("/user")
```
해당부분에서 복수형-> 단수형으로변경하신 이유가있을까요 ?👀

rest api 설계시 endpoint 설계에 관련한 포스트링크를 첨부합니다. :)
https://blog.pumpkin-raccoon.com/115

wsh096 replied:
저번 주에 크게 팀내에서 API 명세서를 통일하는 작업을 진행했습니다. 해당 형태로 고정하고 가자는 의견이었고,
저희도 부분적으로 수정이 이루어지면서 단수 > 복수 > 단수로 되돌아 오는 과정을 거쳤는데요.

현재는 개발 API를 픽스하는 시점이 되어, 복수형으로 사용해도 좋을 것으로 보이는 요소들도 단수로 통일적으로 만들자가 팀내 룰이 되어 적용하게 되었습니다. :)

자세한 건 [API 명세서](https://www.notion.so/API-4586c929e90b49edb6cd024918485c81)를 참고해 주시면 될 것 같습니다.

FeelingXD commented on src/main/java/com/zerobase/yogizogi/user/controller/UserSignUpController.java:
```diff
@@ -11,18 +11,18 @@
    private final UserSignUpService userSignUpService;
 
-   @PostMapping("/signup")
+   @PostMapping("/sign-up")
```
대쉬를사용하여 조금더 직관적이도록 작성하셔서 잘한것 같습니다.

wsh096 replied:
감사합니다. :) 제가 한 건 아니고, 팀 내에 같이 계셨던...(과거형인게 슬픈) API 명세서 작업을 같이 통합적으로 해주신 분이 추가해 주셨던 요소인데 역시는 역시인가 봅니다(?)

FeelingXD commented on src/main/java/com/zerobase/yogizogi/user/service/UserSignUpService.java:
```diff
@@ -45,24 +50,40 @@ public String signUp(UserSignUpForm userSignUpForm) {
+        message.put("msg", "인증 메일 발송에 성공했습니다.");
 
-        return "users/active";
+        return new ApiResponse<>(
+            ResponseCode.RESPONSE_SUCCESS.getCode(),
+            HttpStatus.OK,
+            "SUCCESS",
+            message
+        );
```
서비스 레이어 에서 ApiResponse반환은 권장드리지않습니다. 🤔
또 ApiResponse의 Builder가아닌 생성자를 사용 하신 이유가있을까요 ?

서비스 레이어의책임은 data의 생성 까지만 책임지도록 작성하는걸 권장드리고 싶습니다.

wsh096 replied:
요 부분이 멘토링 받으면서도, 오늘 이야기가 나왔습니다. :) 
서비스는 Controller의 요청을 받아서 처리하는 더 포괄적인 개념의 클래스라는 부분으로 가이드를 받았고, 
더불어서 정적 팩토리 메서드 형태와  Controller에서 처리해줄 것을 이야기 해주셔서 해당 형태가 될 것 같습니다. :) 
이전에도 한 번 이야기 해주셨는데(ResponseEntity) 형태로, Controller에서 보내는 형태가 되지 않을까 싶습니다. :) 

제가 능력이 부족해서, Service에 해당 것들을 매칭하고 반환하고, Controller를 작게 설계하는 잘못된 습관이 들어 있었던게 이렇게 딱 고쳐지는 거 같습니다 :)

FeelingXD reviewed (commented):
항상 열심히하시는군요 몇가지 리뷰를 작성했습니다. :)

wsh096 committed:
fix : 🚑 jwt 토큰에 대한 학습, 토큰값 관련 변경 로직. boolean 관계.

1. jwt 토큰 로직 관련 36번째 JwtAuthenticationProvider 참고.

2. !역으로 되어 있던 관계, -> !을 푼 관계로 변경.(모두 적용 완료)

3. Aes256Utils 사용하지 않아 삭제.

4.Accommodation 관련 controller/service 필요하지 않아 삭제

5.from Dto에 생성해 진행

6. UserController, UserService로 통일.(toEntity) 형태로 respose변경 예정

7. 현재 toEntity가 잘 되지 않아 헤매는 중 ㅠ

BookController 위치

8. ResponseCode SUCCESS 외에는 여기서 쓰는 게 없기 때문에, SUCCESS 만 사용 가능하게 남김

9. repository 에서 _를 이용해 해당 테이블의 원하는 값을 가져오는 방식으로 repository 로직 변경(멘토링 내용 반영)

10. Data 삭제 (ResponseAPI -> toString() 구문도 사용하지 않아 삭제 )

wsh096 committed:
fix : 🚑 response 응답값 형태 수정 중 메일
 send logic

1. 실제 서비스 가입에서 이메일 검증 간단하게 @유무로. 다만, 현재 서비스는 가입 클릭을 눌렀을 때 해당 로직이 작동하는 문제가 있음.(개별적 확인 형태 X)

2. Getter/Setter/생성자 필요한 것만 사용 중.

3. 가독성 개선 중

4. 이메일 보내는 로직 위치에 따른 서비스 적합성 형태 서비스 시 위에(코드리뷰 때 부연 예정)

5. 이메일 인증을 통한 검증 활성화시 리다이렉트 주소 형태의 접근 방식 필요할 것으로 보임(전체 회의에서 소통)

wsh096 committed:
fix : 🚑 response 응답값 형태 수정 중 메일
 send logic

1. 실제 서비스 가입에서 이메일 검증 간단하게 @유무로. 다만, 현재 서비스는 가입 클릭을 눌렀을 때 해당 로직이 작동하는 문제가 있음.(개별적 확인 형태 X)

2. Getter/Setter/생성자 필요한 것만 사용 중.

3. 가독성 개선 중

4. 이메일 보내는 로직 위치에 따른 서비스 적합성 형태 서비스 시 위에(코드리뷰 때 부연 예정)

5. 이메일 인증을 통한 검증 활성화시 리다이렉트 주소 형태의 접근 방식 필요할 것으로 보임(전체 회의에서 소통)

6.포맷 정리 완료

7.OAuth 클래스도 하나로 묶음.

8.서비스 책임을 최소화

9. 불필요 패키지, 클래스 전체 정리.

wsh096 committed:
fix : 🚑 code 리뷰 하며 정리
 1.UserResultDto 삭제.

 2. reviewUpdate method 일반화

 3. ApiResponse _ build 사용하지 않기 때문에 해당 메서드 삭제.

 4. 기타 불필요 어노테이션 삭제.

wsh096 committed:
fix : 🚑 bookList 팩토리 메서드 수정, (빈값들 넣어줌)

1. ReviewList Getter로 직렬화

2. 평점. review 변경으로 수정되는 로직, 내용, 일부 수정.

wsh096 committed:
fix : 🚑 todo delete update 필요

wsh096 committed:
fix : 🚑 reviewDelete  TODO 완료

1. 리뷰 삭제시, 값 변경
(단, 모두 삭제시 기존 값으로 유지 됨._현재데이터들이 리뷰를 다 가져오지 않기 때문에 생기는 문제.)

wsh096 committed:
fix : 🚑 reviewDelete  TODO 완료

1. 리뷰 삭제시, 값 변경
(단, 모두 삭제시 기존 값으로 유지 됨._현재데이터들이 리뷰를 다 가져오지 않기 때문에 생기는 문제.)

2. 리뷰 작성은 사용이 완료된 이후 부터 가능  NOT_ALLOW_WRITE_REVIEW(HttpStatus.BAD_REQUEST, "리뷰는 사용이 완료된 후 작성이 가능합니다."),

3. 예약이 삭제되면 roomCnt 값을 올림.

wsh096 committed:
fix : 🚑 db 비밀번호 업데이트.

bookList 역순으로 나오게 만들기.
price null인 경우의 예외처리(정확히는 ,날짜가 null인 경우의 예외처리 추가)
